server/rest/src/main/java/org/infinispan/rest/server/operations/CacheOperations.java

```diff
@@ -0,0 +1,303 @@
+      try {
+         String cacheName = request.getCacheName();
+         AdvancedCache<String, byte[]> cache = restCacheManager.getCache(cacheName);
+         CacheSet<String> keys = cache.keySet();
+         MediaType mediaType = request.getMediaType()
+               .map(m -> MediaType.fromMediaTypeAsString(m))
+               .orElseThrow(NoMediaTypeException::new);
+
+         InfinispanResponse response = InfinispanResponse.inReplyTo(request);
+         response.contentAsBytes(mediaType.getOutputPrinter().print(cacheName, keys));
+         return response;
+      } catch (CacheException cacheException) {
+         throw new NoCacheFoundException(cacheException.getLocalizedMessage());
+      }
+   }
+
+   public InfinispanResponse getCacheValue(InfinispanRequest request) throws RestResponseException {
+      try {
+         String cacheName = request.getCacheName();
+         String key = request.getKey().orElseThrow(NoKeyException::new);
+         String cacheControl = request.getCacheControl().orElse("");
+         boolean returnBody = request.getRawRequest().method() == HttpMethod.GET;
+         CacheEntry<String, byte[]> entry = restCacheManager.getInternalEntry(cacheName, key);
+         MediaType mediaType = request.getMediaType()
+               .map(m -> MediaType.fromMediaTypeAsString(m))
+               .orElse(MediaType.TEXT_PLAIN);
+
+         InfinispanResponse response = InfinispanResponse.inReplyTo(request);
+         response.status(HttpResponseStatus.NOT_FOUND);
+
+         if (entry instanceof InternalCacheEntry) {
+            InternalCacheEntry<String, byte[]> ice = (InternalCacheEntry<String, byte[]>) entry;
+            Date lastMod = lastModified(ice);
+            Date expires = ice.canExpire() ? new Date(ice.getExpiryTime()) : null;
+            OptionalInt minFreshSeconds = minFresh(cacheControl);
+            if (entryFreshEnough(expires, minFreshSeconds)) {
+               Metadata meta = ice.getMetadata();
+               Optional<String> clientEtag = request.getEtag();
+               // The item already exists in the cache, evaluate preconditions based on its attributes and the headers
+               String etag = calcETAG(ice.getValue(), ((MimeMetadata) meta).contentType());
+               if (clientEtag.isPresent() && clientEtag.get().equals(etag)) {
+                  //client's and our ETAG match. Nothing to do, an entry is cached on the client side...
+                  response.status(HttpResponseStatus.NOT_MODIFIED);
+                  return response;
+               }
+
+               response.status(HttpResponseStatus.OK);
+               response.lastModified(lastMod);
+               response.cacheControl(calcCacheControl(expires));
+               response.expires(expires);
+               response.timeToLive(meta.lifespan());
+               response.maxIdle(meta.maxIdle());
+
+               if (request.getExtended().isPresent() && supportsExtendedHeaders(request.getExtended().get())) {
```

Comment:
No it is fine, just seems odd to use a query param and not actually provide a value for the param. This may be standard usage, just seems odd to me.